ConnectorRhino/ConnectorRhino/ConnectorRhinoShared/UI/ConnectorBindingsRhino.Receive.cs

```diff
@@ -314,26 +314,29 @@ public override async Task<StreamState> ReceiveStream(StreamState state, Progres
   // gets objects by id directly or by application id user string
   private List<RhinoObject> GetObjectsByApplicationId(string applicationId)
   {
+    List<RhinoObject> match = new();
+
     if (string.IsNullOrEmpty(applicationId))
     {
-      return new List<RhinoObject>();
+      return match;
     }
 
     // first try to find the object by app id user string
-    var match = Doc.Objects.FindByUserString(ApplicationIdKey, applicationId, true).ToList();
+    if (Doc.Objects.FindByUserString(ApplicationIdKey, applicationId, true) is RhinoObject[] foundObjects)
+    {
+      match = foundObjects.ToList();
+    }
 
     // if nothing is found, look for the geom obj by its guid directly
-    if (!match.Any())
+    if (match.Count == 0)
     {
-      try
+      if (Utils.GetGuidFromString(applicationId, out Guid id))
       {
-        RhinoObject obj = Doc.Objects.FindId(new Guid(applicationId));
-        if (obj != null)
+        if (Doc.Objects.FindId(id) is RhinoObject obj)
         {
           match.Add(obj);
         }
       }
```

Comment:
Do we know why we might have wanted to catch here? Not seeing anything obvious?

Reply:
Same as [this comment](https://github.com/specklesystems/speckle-sharp/pull/3110#discussion_r1427382698)